src/components/RichTextEditor/RichTextEditor.tsx

```diff
@@ -1,98 +1,73 @@
-import EditorJS, { LogLevels, OutputData } from "@editorjs/editorjs";
+import { LogLevels, OutputData } from "@editorjs/editorjs";
 import { FormControl, FormHelperText, InputLabel } from "@material-ui/core";
-import { PromiseQueue } from "@saleor/misc";
+import { useId } from "@reach/auto-id";
+import { Props as ReactEditorJSProps } from "@react-editor-js/core";
 import classNames from "classnames";
 import React from "react";
+import { createReactEditorJS } from "react-editor-js";
 
-import { RichTextEditorContentProps, tools } from "./RichTextEditorContent";
+import { tools } from "./consts";
+import { useHasRendered } from "./hooks";
 import useStyles from "./styles";
-import { clean } from "./utils";
 
-export type RichTextEditorChange = (data: OutputData) => void;
-export interface RichTextEditorProps extends RichTextEditorContentProps {
+export type EditorJsProps = Omit<ReactEditorJSProps, "factory">;
+
+// https://github.com/Jungwoo-An/react-editor-js#how-to-access-editor-js-instance
+export interface EditorCore {
+  destroy(): Promise<void>;
+  clear(): Promise<void>;
+  save(): Promise<OutputData>;
+  render(data: OutputData): Promise<void>;
+}
+
+export interface RichTextEditorProps extends Omit<EditorJsProps, "onChange"> {
+  id?: string;
   disabled: boolean;
   error: boolean;
   helperText: string;
   label: string;
   name: string;
-  onChange: RichTextEditorChange;
+  editorRef:
+    | React.RefCallback<EditorCore>
+    | React.MutableRefObject<EditorCore>
+    | null;
+  // onChange with value shouldn't be used due to issues with React and EditorJS integration
+  onChange?: () => void;
```

Comment:
It's used to show modals when leaving form with unsaved changes
<img width="853" alt="CleanShot 2022-05-23 at 10 18 48@2x" src="https://user-images.githubusercontent.com/4144459/169775732-2025ab7e-f3df-4ccc-a53b-8b8e402c0142.png">


we don't want it `onBlur` - because that would be just when user leaves the input (they don't have to change it) so we have `onChange` but without any value from EditorJS